sphinx/pycode/ast.py

```diff
@@ -202,6 +202,9 @@ def is_simple_tuple(value: ast.AST) -> bool:
             return "%s[%s]" % (self.visit(node.value), self.visit(node.slice))
 
     def visit_UnaryOp(self, node: ast.UnaryOp) -> str:
+        # UnaryOp is one of {UAdd, USub, Invert, Not}. Only Not needs a space.
+        if not isinstance(node.op, ast.Not):
+            return "%s%s" % (self.visit(node.op), self.visit(node.operand))
         return "%s %s" % (self.visit(node.op), self.visit(node.operand))
```

Comment:
I consider `ast.Not` is a special case. So it would be better to check it first.
```suggestion
        # UnaryOp is one of {UAdd, USub, Invert, Not}. Only Not needs a space.
        if isinstance(node.op, ast.Not):
            return "%s %s" % (self.visit(node.op), self.visit(node.operand))
        else:
            return "%s%s" % (self.visit(node.op), self.visit(node.operand))
```